[Feature] likesCount 기능 구현

ghrltjdtprbs opened the pull request:
## [Feature] likesCount 기능 구현
수정사항 :
- LikeTripService의 toggleLike 메서드 내에서 likesCount를 업데이트하는 로직을 추가였습니다.
- 사용자가 좋아요를 추가하면 likesCount를 증가시키고, 좋아요를 해제하면 감소합니다.

ghrltjdtprbs committed:
[feat] likesCount 구현

수정사항 :
- LikeTripService의 toggleLike 메서드 내에서 likesCount를 업데이트하는 로직을 추가.
- 사용자가 좋아요를 추가하면 likesCount를 증가시키고, 좋아요를 해제하면 감소

ghrltjdtprbs committed:
[refactor] likesCount 에러 해결

문제 사항 :
- 좋아요 반복 취소시 likesCount가 음수로 나오는 에러발견.
수정 사항:
- likesCount 최소값을 0으로 설정하였습니다.

ghrltjdtprbs committed:
[refactor] LikeTripService 주석제거

수정 사항:
- LikeTripService 주석을 제거하였습니다.

ghrltjdtprbs committed:
Merge branch 'develop' of https://github.com/FC-BE-ToyProject-Team6/KDT_Y_BE_Toy_Project3_DEV into feature/liketrip

sungjiwoon commented on src/main/java/com/fastcampus/toyproject/domain/liketrip/service/LikeTripService.java:
```diff
@@ -12,31 +12,47 @@
+
     private final LikeTripRepository likeTripRepository;
     private final TripRepository tripRepository;
     private final UserService userService;
```
tripService로 선언 부탁드려용!!

ghrltjdtprbs replied:
tripRepository -> tripService로 변경 부분은 다음 pr로 나누려고 했습니다! 그냥 바로 커밋할게요 ㅎㅎ👍

sungjiwoon commented on src/main/java/com/fastcampus/toyproject/domain/liketrip/service/LikeTripService.java:
```diff
@@ -12,31 +12,47 @@
         User user = userService.getUser(userId);
         Trip trip = tripRepository.findById(request.getTripId())
             .orElseThrow(() -> new TripException(TripExceptionCode.NO_SUCH_TRIP));
 
```
위에 user 처럼 tripService에서 가져오시면 될것 같애용!

ghrltjdtprbs replied:
새로운 커밋 확인해주세요! ㅎㅎ

sungjiwoon commented on src/main/java/com/fastcampus/toyproject/domain/trip/entity/Trip.java:
```diff
@@ -79,6 +79,7 @@ public class Trip {
     private Boolean isDomestic;
 
+    @ColumnDefault("0")
     @Comment("좋아요 개수")
```
👏

ghrltjdtprbs committed:
[refactor] LikeTripService 수정(피드백 반영)

문제 사항:
- LikeTripService에서 사용하던 tripRepository 대신 tripService 사용할 수 있도록 수정 요청.

수정 사항:
- tripRepository -> tripService
- updateLikesCount메소드를 LikeTripService -> TripService로 이동

sungjiwoon reviewed (approved):
수고하셨습니당!
